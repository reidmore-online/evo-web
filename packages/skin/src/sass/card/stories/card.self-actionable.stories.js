export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
