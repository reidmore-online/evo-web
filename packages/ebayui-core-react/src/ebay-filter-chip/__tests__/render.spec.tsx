import React from "react";
import { composeStories } from "@storybook/react-vite";

import * as stories from "./index.stories";
import { render } from "@testing-library/react";

const {
    Default,
    Selected,
    WithIcon,
    Expressive,
    ExpressiveSelected,
    Menu,
    MenuExpanded,
    Anchor,
    AnchorSelected,
    Disabled,
    Collection,
} = composeStories(stories);

describe("<EbayFilterChip /> rendering", () => {
    it("renders default story correctly", () => {
        const { container } = render(<Default />);
        expect(container).toMatchSnapshot();
    });

    it("renders selected story correctly", () => {
        const { container } = render(<Selected />);
        expect(container).toMatchSnapshot();
    });

    it("renders with icon story correctly", () => {
        const { container } = render(<WithIcon />);
        expect(container).toMatchSnapshot();
    });

    it("renders expressive story correctly", () => {
        const { container } = render(<Expressive />);
        expect(container).toMatchSnapshot();
    });

    it("renders expressive selected story correctly", () => {
        const { container } = render(<ExpressiveSelected />);
        expect(container).toMatchSnapshot();
    });

    it("renders menu story correctly", () => {
        const { container } = render(<Menu />);
        expect(container).toMatchSnapshot();
    });

    it("renders menu expanded story correctly", () => {
        const { container } = render(<MenuExpanded />);
        expect(container).toMatchSnapshot();
    });

    it("renders anchor story correctly", () => {
        const { container } = render(<Anchor />);
        expect(container).toMatchSnapshot();
    });

    it("renders anchor selected story correctly", () => {
        const { container } = render(<AnchorSelected />);
        expect(container).toMatchSnapshot();
    });

    it("renders disabled story correctly", () => {
        const { container } = render(<Disabled />);
        expect(container).toMatchSnapshot();
    });

    it("renders collection story correctly", () => {
        const { container } = render(<Collection />);
        expect(container).toMatchSnapshot();
    });
});
