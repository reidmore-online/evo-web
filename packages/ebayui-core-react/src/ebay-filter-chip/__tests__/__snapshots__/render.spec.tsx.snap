// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<EbayFilterChip /> rendering renders anchor selected story correctly 1`] = `
<div>
  <a
    class="filter-chip filter-chip--selected filter-chip--animated"
    href="https://ebay.com"
  >
    <span
      class="filter-chip__text"
    >
      Selected Link
      <span
        class="clipped"
      >
        - 
        Filter Applied
      </span>
    </span>
  </a>
</div>
`;

exports[`<EbayFilterChip /> rendering renders anchor story correctly 1`] = `
<div>
  <a
    class="filter-chip filter-chip--animated"
    href="https://ebay.com"
  >
    <span
      class="filter-chip__text"
    >
      Link Filter
    </span>
  </a>
</div>
`;

exports[`<EbayFilterChip /> rendering renders collection story correctly 1`] = `
<div>
  <div
    style="display: flex; gap: 8px; flex-wrap: wrap;"
  >
    <button
      aria-pressed="false"
      class="filter-chip filter-chip--animated"
      type="button"
    >
      <span
        class="filter-chip__text"
      >
        Brand
      </span>
    </button>
    <button
      aria-pressed="true"
      class="filter-chip filter-chip--animated"
      type="button"
    >
      <span
        class="filter-chip__text"
      >
        Color
      </span>
    </button>
    <button
      aria-pressed="false"
      class="filter-chip filter-chip--animated"
      type="button"
    >
      <span
        class="filter-chip__text"
      >
        Size
      </span>
    </button>
    <button
      aria-expanded="false"
      aria-pressed="false"
      class="filter-chip filter-chip--animated"
      type="button"
    >
      <span
        class="filter-chip__text"
      >
        Price Range
      </span>
      <svg
        aria-hidden="true"
        class="filter-chip__trailing icon icon--12"
        focusable="false"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-chevron-down-12"
        />
      </svg>
    </button>
    <button
      aria-pressed="false"
      class="filter-chip filter-chip--animated"
      disabled=""
      type="button"
    >
      <span
        class="filter-chip__text"
      >
        Out of Stock
      </span>
    </button>
  </div>
</div>
`;

exports[`<EbayFilterChip /> rendering renders default story correctly 1`] = `
<div>
  <button
    aria-pressed="false"
    class="filter-chip filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Default Filter
    </span>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders disabled story correctly 1`] = `
<div>
  <button
    aria-pressed="false"
    class="filter-chip filter-chip--animated"
    disabled=""
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Disabled Filter
    </span>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders expressive selected story correctly 1`] = `
<div>
  <button
    aria-pressed="true"
    class="filter-chip filter-chip--expressive filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__media"
    >
      <img
        alt="Category"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png"
      />
    </span>
    <span
      class="filter-chip__text"
    >
      Expressive Selected
    </span>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders expressive story correctly 1`] = `
<div>
  <button
    aria-pressed="false"
    class="filter-chip filter-chip--expressive filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__media"
    >
      <img
        alt="Category"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png"
      />
    </span>
    <span
      class="filter-chip__text"
    >
      Expressive Filter
    </span>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders menu expanded story correctly 1`] = `
<div>
  <button
    aria-expanded="true"
    aria-pressed="false"
    class="filter-chip filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Menu Expanded
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
      focusable="false"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-chevron-down-12"
      />
    </svg>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders menu story correctly 1`] = `
<div>
  <button
    aria-expanded="false"
    aria-pressed="false"
    class="filter-chip filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Menu Filter
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
      focusable="false"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-chevron-down-12"
      />
    </svg>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders selected story correctly 1`] = `
<div>
  <button
    aria-pressed="true"
    class="filter-chip filter-chip--animated"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Selected Filter
    </span>
  </button>
</div>
`;

exports[`<EbayFilterChip /> rendering renders with icon story correctly 1`] = `
<div>
  <button
    aria-pressed="false"
    class="filter-chip filter-chip--animated"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="icon icon--16"
      focusable="false"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-sneaker-16"
      />
    </svg>
    <span
      class="filter-chip__text"
    >
      Filter with Icon
    </span>
  </button>
</div>
`;
